pandas/tests/reshape/merge/test_join.py

```diff
@@ -228,16 +228,19 @@ def test_join_on_fails_with_different_column_counts(self):
                             index=tm.makeCustomIndex(10, 2))
             merge(df, df2, right_on='a', left_on=['a', 'b'])
 
-    def test_join_on_fails_with_wrong_object_type(self):
-        # GH12081
-        wrongly_typed = [Series([0, 1]), 2, 'str', None, np.array([0, 1])]
-        df = DataFrame({'a': [1, 1]})
+    @pytest.mark.parametrize("wrong_type", [2, 'str', None, np.array([0, 1])])
+    def test_join_on_fails_with_wrong_object_type(self, wrong_type):
+        # GH12081 - original issue
+
+        # GH21220 - merging of Series and DataFrame is now allowed
+        # Edited the test to remove the Series object from test parameters
+        # Also, parameterized the original test
```

Comment:
this last line of the comment is not really needed (in the sense of: it is not really useful for future readers of this code, they will see that the test is parametrized)

Reply:
Thanks - makes sense, will remove the redundant comment - will make the change together with any changes needed for your ValueError/TypeError comment above.